src/pages/recipes/events/index.md

```diff
@@ -0,0 +1,139 @@
+let myEventListener = app.addEventListener("afterNew", myDisplayEventType);
+```
+The preceding script fragment refers to the following function:
+
+```js
+function myDisplayEventType(myEvent){
+    alert("This event is the " + myEvent.eventType + " event.");
+}
+```
+
+**Removing an eventListener**
+
+To remove the eventListener created by the preceding script, run the following script:
+
+```js
+let myInDesign = require("indesign");
+let app = myInDesign.app;
+let myResult = app.removeEventListener("afterNew", myDisplayEventType);
+```
+When an eventListener responds to an event, the event may still be processed by other eventListeners that might be monitoring the event (depending on the propagation of the event). For example, the afterOpen event can be observed by eventListeners associated with both the application and the document.
+
+eventListeners do not persist beyond the current InDesign session. To make an eventListener available in every InDesign session, add the script to the startup scripts folder. 
+
+An event can trigger multiple eventListeners as it propagates through the scripting object model. The following sample script demonstrates an event triggering eventListeners registered to different objects:
+
+```js
+let myInDesign = require("indesign");
+let app = myInDesign.app;
+let myApplicationEventListener = app.eventListeners.add("beforeImport", myEventInfo);
+let myDocumentEventListener = app.documents.item(0).eventListeners.add("beforeImport", myEventInfo);
+function myEventInfo(myEvent){
+    let myString = "Current Target: " + myEvent.currentTarget.name;
+    alert(myString);
+}
+```
+When you run the preceding script and place a file, InDesign displays following information in sequence, and as alerts:
+1. Document name
+2. Application name
+
+Use the following script to remove the event listeners added by the preceding script:
+
+```js
+let myInDesign = require("indesign");
+let app = myInDesign.app;
+app.removeEventListener("beforeImport", myEventInfo);
+app.documents.item(0).removeEventListener("beforeImport", myEventInfo);
+```
+
+## Writing Events Script to Run Through Out the Session
+After the event listener is associated, the script is blocked by a promise and wait for the event in the background until the event listener is removed. If the event occurs, there is an automatic callback to execute the related function. If the event listener is removed, the promise can be resolved to complete the script execution.
+
+Here is a sample events script to run and listen for “afterNew” event throughout the InDesign session and prints a relevant string whenever a new document is created.
+
+```js
+let myInDesign = require("indesign");
+let app = myInDesign.app;
+ 
+await main();
+ 
+async function main()
+{
+  return new Promise((resolve, reject) => { mySnippet(); })
+}
+ 
+function mySnippet()
+{
+    //![Add event listener.]
+    let myEventListener = app.addEventListener("afterNew", myDisplayEventType);
```

Comment:
Why do we need myEventListener?

Reply:
It is an eventListener which triggers the mDisplayEventType() function as soon as the "afterNew" event takes place.

Reply:
Needs review.

Reply:
I think we can replace the line with just `app.addEventListener("afterNew", myDisplayEventType);` We dont need the variable since we dont use it anywhere.